### PR TITLE
Changes snack food insertion

### LINKED ISSUE #8187: Apparently pizzas can hold an unlimited number of items which are all released in a huge, ridiculous pile when the pizza is sliced
I didn't think to screencap it but someone did personally show this to me so I know it happens

Also this:
image

### LINKED ISSUE #9224: FRACTAL WATERMELON
image

### PR DESCRIPTION
**Pull request by clusterfack**
Probably fixes like 3-4 issues on this topic

Fixes #9224
Fixes #8187
Namely
1. A food object being qdeleted drops the contents of it on the ground
2. A food object cannot hold another of the same size as it (must be of lower size)
3. A food object can only hold 1/2 of its slices_num (number of slices when sliced)

### CODE REVIEW

**Comment by Intigracy** [comment]
9224
8187

Please undo change 4, you can't slice slices of pizza to remove the items.
Just make them pop out like they currently do.

**Comment by clusterfack** [comment]
You can eat them and the items would pop out of the slices

**Comment by Intigracy** [comment]
But the slices are tiny compared to the entire cheese wheel, per say.

It just makes more sense to pop the items out on slice.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -230,6 +242,8 @@
 	var/reagents_per_slice = reagents.total_volume/slices_num //Figure out how much reagents each slice inherits (losing slices loses reagents)
 	for(var/i = 1 to (slices_num - slices_lost)) //Transfer those reagents
 		var/obj/slice = new slice_path(src.loc)
+		if(contents.len)
+			slice.contents += contents[1] //Our contents move to the slices
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Won't it just add the same object to every one of the slices?

**Comment by clusterfack** [comment]
It seems people think it is ridiculous though

>Apparently pizzas can hold an unlimited number of items which are all released in a huge, ridiculous pile when the pizza is sliced #8187

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Why in the world would you think that?

Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: I don't see where it's removing contents[1] from contents, it's just adding it to slice.contents.

**Comment by Intigracy** [comment]
Yeah, the problem is the unlimited number of items more than it is the dumped out when sliced.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Don't you know that contents is internally maintained?

Adding contents[1] to another items contents is the equivalent of setting the loc of contents[1] to the new object, which internally results in the removal of contents[1] from contents.

**Comment by clusterfack** [comment]
Doesn't read that way to me

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: did not know, thanks fam

**Comment by Exxion** [comment]
I know I'm a cool dude, but my word shouldn't be taken quite that seriously

My biggest problem is the unlimited size, but I think it's fine with or without change 4

**Comment by Probe1** [comment]
>unlimited items in a food obj

who the fuck coded this, goofball?

I BET IT WAS

**Comment by Iamgoofball** [comment]
wasnt me this time

**Comment by Iamgoofball** [comment]
but it sure sounds like something id code brb adding it to the list

**Comment by Probe1** [comment]
YOU

**Comment by clusterfack** [comment]
IT WAS ME, CLUSTER!

**Pull request merged**